### DISCUSSION TITLE
Confusion about C # language documents

### DISCUSSION THREAD

**Discussion by tylike** [dotnet/csharplang#7088] [Q&A] [answered]
I plan to develop the project using Dotnet7(C#11.0)+VS2022(17.5.3), which requires the use of DIM (default interface method) language features.

I can only find some examples in the relevant documents of c # 8.0.

For example: 
var helpdocument = 
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-8.0/default-interface-methods

On this page, I copied the following code:
```cs
interface IA
{
    void M() { WriteLine("IA.M"); }
}
interface IB : IA
{
    override void IA.M() { WriteLine("IB.M"); } // explicitly named   **CS0106 :The modifier 'override ' is not valid for this item**
}
interface IC : IA
{
    override void M() { WriteLine("IC.M"); } // implicitly named    **CS0106 :The modifier 'override ' is not valid for this item**
}
```
My csproj file is as follows:
```xml
<Project Sdk="Microsoft.NET.Sdk">
	<PropertyGroup>
		<OutputType>Exe</OutputType>
		<TargetFramework>net7.0</TargetFramework>
		<ImplicitUsings>disable</ImplicitUsings>
		<Nullable>enable</Nullable>
	</PropertyGroup>
</Project>
```
Visual studio prompted me with the following error:
**CS0106 :The modifier 'override ' is not valid for this item**
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/cs0106?f1url=%3FappId%3Droslyn%26k%3Dk(CS0106)

I guess the syntax above was valid when writing this document

**My question is: I want to use the dim feature under c # 11.0. Where can I get the syntax specification for the Default interface method in c # 11.0 version?**

The helpdocument lists many examples, some of which may have been invalidated in c # 11.0, which makes me very confused.
Who can help me, thank you!

**Comment by En3Tho** [comment, marked as answer]
You were looking at proposal document, not the one describing implemented design. I've stumbled upon this too, once. Googling dims actually falsely leads to proposal document.

Try this one. Note that implemented design has much narrower scope of what is allowed compared to that proposal doc.

https://learn.microsoft.com/en-us/dotnet/csharp/advanced-topics/interface-implementation/default-interface-methods-versions

**Reply by TahirAhmadov** [reply]
This link is broken...

**Reply by En3Tho** [reply]
Thanks. Fixed.
